src/Compilers/Core/Portable/Text/SubText.cs

```diff
@@ -89,5 +94,85 @@ private TextSpan GetCompositeSpan(int start, int length)
             int compositeEnd = Math.Min(UnderlyingText.Length, compositeStart + length);
             return new TextSpan(compositeStart, compositeEnd - compositeStart);
         }
+
+        /// <summary>
+        /// Delegates to the SubText's UnderlyingText to determine line information.
+        /// </summary>
+        internal sealed class SubTextLineInfo : TextLineCollection
+        {
+            private readonly SubText _subText;
+            private readonly int _startLineNumber;
+            private readonly int _lineCount;
+            private readonly bool _startsWithinSplitCRLF;
+            private readonly bool _endsWithinSplitCRLF;
+
+            public SubTextLineInfo(SubText subText)
+            {
+                _subText = subText;
+
+                var startLine = _subText.UnderlyingText.Lines.GetLineFromPosition(_subText.UnderlyingSpan.Start);
+                var endLine = _subText.UnderlyingText.Lines.GetLineFromPosition(_subText.UnderlyingSpan.End);
+
+                _startLineNumber = startLine.LineNumber;
+                _lineCount = (endLine.LineNumber - _startLineNumber) + 1;
+
+                if (_subText.UnderlyingSpan.Length > 0)
+                {
+                    if (startLine.End == _subText.UnderlyingSpan.Start - 1 && startLine.EndIncludingLineBreak == _subText.UnderlyingSpan.Start + 1)
+                    {
+                        _startsWithinSplitCRLF = true;
```

Comment:
i don't quite get how you know this stuff is true.

Reply:
I re-ordered and re-formatted the conditions, maybe that will make it seem a bit less muddled?